### PR TITLE
Update TaskDetails and TaskInfo for v0.30.0

### PR DESCRIPTION
**Pull request by alallema**
V0.30.0
in TasksInfo:
  - [x] Remove `duration`
  - [x] Remove `startedAt`
  - [x] Remove `finishedAt`
  - [x] Remove `error`
  - [x] Remove `details`

in TaskDetais:
  - [x] Add `Pagination` and create `Pagination` class
  - [x] Add `Faceting` and create `Faceting` class
  - [x] Add `providedIds`
  - [x] Add `dumpUid`
  - [x] Add `matchedTasks`
  - [x] Add `originalFilter`
  - [x] Add `canceledTasks`
  - [x] Add `deletedTasks`

Creation of the `Pagination` and the `Faceting` class.

### CODE REVIEW

File: src/main/java/com/meilisearch/sdk/model/TaskInfo.java [hunk 1]
```diff
@@ -15,12 +14,7 @@ public class TaskInfo {
     protected int taskUid = 0;
     protected String indexUid = "";
     protected String type = null;
-    protected String duration = "";
```
Comment (brunoocasali) [inline src/main/java/com/meilisearch/sdk/model/TaskInfo.java:18]: Were those attributes moved to "Task"/"TaskDetails"?

Comment (alallema) [reply src/main/java/com/meilisearch/sdk/model/TaskInfo.java:18]: They have not been moved, they are still in `Task` but from the v0.30.0 there are only these 4 fixed fields left `taskUid`, `indexUid`, `type`, `enqueuedAt`.

Comment (brunoocasali) [reply src/main/java/com/meilisearch/sdk/model/TaskInfo.java:18]: Nice, cool!

**Review by brunoocasali — APPROVED** [review]
LGTM! 🍖

**Pull request merged**